jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsRequestHandler.java

```diff
@@ -87,11 +92,16 @@ public JaxRsRequestHandler(Object[] services,
 
         long requestStartTime = System.currentTimeMillis();
 
-        return resource.call(jaxRsRequest)
-            .singleOrDefault(null)
-            .flatMap(result -> writeResult(response, result))
-            .onErrorResumeNext(e -> exceptionHandler.handleException(request, response, e))
-            .doAfterTerminate(() -> resource.log(request, response, requestStartTime));
+        try {
+            JaxRsRequestContext.open();
+            return resource.call(jaxRsRequest)
+                .singleOrDefault(null)
+                .flatMap(result -> writeResult(response, result))
```

Comment:
If you add a test where writeResult leaves the current thread I think you would find that JaxRsRequestContext.close() would execute before the flatMap terminates.

Reply:
The perils of RxJava. It is true that in the current thread (the thread that creates the observable) the context is closed before the observable executes. The reactive-contexts library solves that by propagating the context to the created observable (and any observables it in turn creates).